src/ddmd/codegenASTFamily.d

```diff
@@ -0,0 +1,56 @@
+module ddmd.codegenASTFamily;
+
+import ddmd.arraytypes;
+import ddmd.dclass;
+import ddmd.denum;
+import ddmd.dimport;
+import ddmd.dstruct;
+import ddmd.expression;
+import ddmd.globals;
+import ddmd.identifier;
+import ddmd.mtype;
+
+struct CodegenASTFamily
+{
+    alias EnumDeclaration = .EnumDeclaration;
+
+    StructDeclaration iStructDeclaration(Loc loc, Identifier id)
+    {
+        return new StructDeclaration(loc, id);
+    }
+
+    UnionDeclaration iUnionDeclaration(Loc loc, Identifier id)
+    {
+        return new UnionDeclaration(loc, id);
+    }
+
+    BaseClass* iBaseClass(Type t)
+    {
+        return new BaseClass(t);
+    }
+
+    InterfaceDeclaration iInterfaceDeclaration(Loc loc, Identifier id, BaseClasses* baseclasses)
+    {
+        return new InterfaceDeclaration(loc, id, baseclasses);
+    }
+
+    ClassDeclaration iClassDeclaration(Loc loc, Identifier id, BaseClasses* baseclasses, Dsymbols* members, bool inObject)
+    {
+        return new ClassDeclaration(loc, id, baseclasses, members, inObject);
+    }
+
+    Import iImport(Loc loc, Identifiers* a, Identifier id, Identifier aliasId, int isstatic)
+    {
+        return new Import(loc, a, id, aliasId, isstatic);
+    }
+
+    EnumDeclaration iEnumDeclaration(Loc loc, Identifier id, Type memtype)
+    {
+        return new EnumDeclaration(loc, id, memtype);
+    }
+
+    EnumMember iEnumMember(Loc loc, Identifier id, Expression value, Type origType)
+    {
+        return new EnumMember(loc, id, value, origType);
+    }
```

Comment:
These forwarding constructors quickly become a lot of boilerplate so we definitely should avoid them.